tensorflow/core/kernels/segment_reduction_ops.cc

```diff
@@ -288,27 +319,65 @@ class UnsortedSegmentSumOp : public OpKernel {
     auto output_flat = output->flat_outer_dims<T>();
 
     auto data_ptr = data.template flat<T>().data();
-    functor::UnsortedSegmentSumFunctor<Device, T, Index>()(
-        context, context->template eigen_device<Device>(), output_rows,
-        segment_ids.shape(), segment_flat, data.NumElements(), data_ptr,
-        output_flat);
+    reductionFunctor(context, context->template eigen_device<Device>(),
+                     output_rows, segment_ids.shape(), segment_flat,
+                     data.NumElements(), data_ptr, output_flat);
   }
+  functor::UnsortedSegmentBaseFunctor<Device, T, Index>& reductionFunctor;
 };
 
-#define REGISTER_CPU_UNSORTED_KERNELS(type, index_type)                \
+template <typename Device, class T, class Index>
+class UnsortedSegmentSumOp : public UnsortedSegmentBaseOp<Device, T, Index> {
+ public:
+  explicit UnsortedSegmentSumOp(OpKernelConstruction* context)
+      : UnsortedSegmentBaseOp<Device, T, Index>(
+            context,
+           *(new functor::UnsortedSegmentSumFunctor<Device, T, Index>())) {}
```

Comment:
wasn't sure how to pass it as reference instead of a pointer as requested.
I changed it now to use a member variable in the derived class (UnsortedSegmentSumOp), to be able to pass the functor (UnsortedSegmentSumFunctor) as a reference to the base class constructor (UnsortedSegmentBaseOp).